code/modules/vending/vending.dm

```diff
@@ -1586,7 +1616,335 @@
 
 		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/food/snacks/breakfast, rand(2, 4), hidden=1)
 		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/food/snacks/snack_cake, rand(1, 3), hidden=1)
+//The burden of these machinations weighs on my shoulders
+//And thus you will be burdened
+/datum/data/vending_product/player_product
+	var/contents = list()
+	var/product_type
+	var/real_name
+	var/image/icon
+	var/label
+	product_amount = 1
+	New(obj/item/product,price)
+		..()
+		product_type = product.type
+		product_name = product.name
+		real_name = product.real_name
+		contents += product
+		product_cost = price
+/obj/item/machineboard
+	name = "machine module"
+	desc = "A circuit board assembly used in the construction of machinery."
+	icon = 'icons/obj/electronics.dmi'
+	icon_state = "board1"
+	mats = 2
+	var/machinepath = null
+/obj/item/machineboard/vending
+	name = "vending machine module"
+	desc = "An assembly used in the construction of a vending machine."
+	machinepath = "/obj/machinery/vending/player"
+	icon = 'icons/obj/vending.dmi'
+	icon_state = "base-module"
+/obj/item/machineboard/vending/player
+	icon_state = "player-module"
+/obj/item/machineboard/vending/monkeys
+	name = "monkey vending module"
+	machinepath = "/obj/machinery/vending/monkey"
+	icon_state = "monkey-module"
+	mats = 0 //No!!
+/obj/machinery/vendingframe
+	name = "vending machine frame"
+	desc = "A generic vending machine frame."
+	icon = 'icons/obj/vending.dmi'
+	icon_state = "standard-frame"
+	density = 1
+	var/wrenched = 0
+	var/glassed = 0
+	var/boardinstalled = 0
+	var/wiresinstalled = 0
+	var/vendingtype = null
+
+	attackby(obj/item/target, mob/user)
+		if(iswrenchingtool(target))
+			playsound(src.loc, "sound/items/Ratchet.ogg", 50, 1)
+			if(!wrenched && do_after(user, 2 SECONDS))
+				wrenched = 1
+				anchored = 1
+				boutput(user, "<span class='notice'>You wrench the frame into place.</span>")
+			else if(!boardinstalled && wrenched && do_after(user, 2 SECONDS))
+				wrenched = 0
+				anchored = 0
+				boutput(user, "<span class='notice'>You unfasten the frame.</span>")
+		else if(istype(target, /obj/item/machineboard/vending))
+			var/obj/item/machineboard/vending/V = target
+			vendingtype = V.machinepath
+			if(wrenched && !boardinstalled)
+				playsound(src.loc, "sound/items/Deconstruct.ogg", 50, 1)
+				icon_state = "standard-frame-electronics"
+				boutput(user, "<span class='notice'>You install the module inside the frame.</span>")
+				user.u_equip(target)
+				target.set_loc(target)
+				boardinstalled = 1
+		else if(istype(target, /obj/item/cable_coil))
+			var/obj/item/cable_coil/targetcoil = target
+			playsound(src.loc, "sound/items/Deconstruct.ogg", 50, 1)
+			if(targetcoil.amount >= 5 && boardinstalled && !wiresinstalled && do_after(user, 2 SECONDS))
```

Comment:
I like playing the sound before the do_after, the other stuff was really inconsistent about it but this is what felt good

Reply:
If board isn't installed, wires are already installed, or coils < 5 it will play the deconstruct sound every time you click it, that is how I found it.

Reply:
Ah! You're right!

Reply:
This is fixed we use an actionbar now so the sound just playes after